coding/javascript.md

```diff
@@ -37,6 +37,26 @@ We use [Yarn](https://yarnpkg.com/en) for dependency management.
 
 We use the [Jest](https://jestjs.io) unittest framework.
 
+#### General Testing Principals
+  * Avoid setting up state in `beforeEach`. It should be clear from reading a test in isolation what state it is in.
+  * Use test factories to reduce boilerplate for commonly instantiated objects.
+  * Don't test external code/libraries.
+
+#### Testing React
+  * Tests should be collocated with their component in the same directory (MyComponent.js should have a corresponding MyComponent.test.js)
```

Comment:
Is this a strong React standard, and not a strong JavaScript one? Because otherwise I think we should choose one structure for all of JavaScript.

Reply:
There aren't any react standards as such for project structure. This is reasonable advice for any modern js, but given we only use react, I think this is okay to live here?

Reply:
I like this approach alot. Makes it easier to quickly find corresponding tests.

Reply:
If it's the advice for JS in general, could we move it into the generic section?

Reply:
I don't thing this can be generic to all JS. Not all JS and it's frameworks are written with a component philosophy. This could be worded as the recommended file structure for a project.